### PR TITLE
OBSDOCS-692: Update Monitoring doc links and topic map position

### PR DESCRIPTION
**Pull request by eromanova97**
Version(s): `enterprise-4.12` and later

Issue: [OBSDOCS-692](https://issues.redhat.com/browse/OBSDOCS-692)

Link to docs preview:  https://74104--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/monitoring/monitoring-overview

QE review:
* N/A

**Additional information:**
This PR changes the topic map file, turns the assembly directories for `monitoring` into subdirectories of `observability`, and updates the affected xrefs.

### CODE REVIEW

File: support/troubleshooting/investigating-monitoring-issues.adoc [hunk 1]
```diff
@@ -17,14 +17,14 @@ include::modules/monitoring-investigating-why-user-defined-metrics-are-unavailab
 [role="_additional-resources"]
 .Additional resources
 
-* xref:../../monitoring/configuring-the-monitoring-stack.adoc#creating-user-defined-workload-monitoring-configmap_configuring-the-monitoring-stack[Creating a user-defined workload monitoring config map]
-* See xref:../../monitoring/managing-metrics.adoc#specifying-how-a-service-is-monitored_managing-metrics[Specifying how a service is monitored] for details on how to create a service monitor or pod monitor
-* See xref:../../monitoring/managing-metrics.adoc#getting-detailed-information-about-a-target_managing-metrics[Getting detailed information about a metrics target]
+* xref:../../observability/monitoring/configuring-the-monitoring-stack.adoc#creating-user-defined-workload-monitoring-configmap_configuring-the-monitoring-stack[Creating a user-defined workload monitoring config map]
```
Comment (nalhadef) [inline support/troubleshooting/investigating-monitoring-issues.adoc:20]: Change "config map" to "configuration map."

**Comment by nalhadef** [comment]
Three small comments.

File: observability/monitoring/enabling-monitoring-for-user-defined-projects.adoc [hunk 2]
```diff
@@ -16,9 +16,9 @@ include::modules/monitoring-enabling-monitoring-for-user-defined-projects.adoc[l
 [role="_additional-resources"]
 .Additional resources
 
-* xref:../monitoring/configuring-the-monitoring-stack.adoc#creating-user-defined-workload-monitoring-configmap_configuring-the-monitoring-stack[Creating a user-defined workload monitoring config map]
-* xref:../monitoring/configuring-the-monitoring-stack.adoc#configuring-the-monitoring-stack[Configuring the monitoring stack]
-* xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#granting-users-permission-to-configure-monitoring-for-user-defined-projects_enabling-monitoring-for-user-defined-projects[Granting users permission to configure monitoring for user-defined projects]
+* xref:../../observability/monitoring/configuring-the-monitoring-stack.adoc#creating-user-defined-workload-monitoring-configmap_configuring-the-monitoring-stack[Creating a user-defined workload monitoring config map]
```
[the remark this answers is not in the document]
Comment (bburt-rh) [reply observability/monitoring/enabling-monitoring-for-user-defined-projects.adoc:19]: FYI, `config map` is a Kubernetes API object term and is used correctly: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references

File: support/troubleshooting/investigating-monitoring-issues.adoc (hunk 1 above)
Comment (bburt-rh) [reply support/troubleshooting/investigating-monitoring-issues.adoc:20]: Same comment as above re `config map`

File: observability/monitoring/configuring-hpa-for-an-application.adoc [hunk 3]
```diff
@@ -15,9 +15,9 @@ include::snippets/technology-preview.adoc[leveloffset=+0]
...
+* Make sure you configured the cluster for application monitoring. In the following example, it is presumed that Prometheus and Alertmanager instances were installed in the `default` project. See xref:../../observability/monitoring/application-monitoring.adoc#configuring-cluster-for-application-monitoring_application-monitoring[Application monitoring].
+* Make sure you configured monitoring for your application. In the following example, it is presumed that the application and the service monitor for it were installed in the `default` project. See xref:../../observability/monitoring/application-monitoring.adoc#configuring-monitoring-for-an-application_application-monitoring[Application monitoring].
+* Make sure your application exposes metrics using the `prometheus-adapter`. In the following example, it is presumed that `prometheus-adapater` was installed in the `default` project. See xref:../../observability/monitoring/application-monitoring.adoc#exposing-custom-application-metrics-for-horizontal-pod-autoscaling_application-monitoring[Application monitoring].
```
[the remark this answers is not in the document]
Comment (eromanova97) [reply observability/monitoring/configuring-hpa-for-an-application.adoc:20]: Oh good catch, thank you!

File: observability/monitoring/enabling-monitoring-for-user-defined-projects.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (eromanova97) [reply observability/monitoring/enabling-monitoring-for-user-defined-projects.adoc:19]: Yes, as Brian already pointed out, this should stay as "config map". Thank you!

**Comment by mburke5678** [comment]
@eromanova97 Everything looks fine to me. I spot-checked some of the updated links, trusting our tools to spot broken links.  Questions before the merge: 

* Has this PR gone through [docs change management](https://docs.google.com/document/d/1fLMpK4bqthtFlCwA36yeo2cI-pExFpX76VMnSi_In6Q/edit#bookmark=id.khm4rxl210qm)? 
Have you [set up redirects](https://docs.google.com/document/d/1fLMpK4bqthtFlCwA36yeo2cI-pExFpX76VMnSi_In6Q/edit#bookmark=id.xb255koulgqx)?

**Comment by bburt-rh** [comment]
@mburke5678 - This is part of a bigger project to move all of the Observability docs in the folder, and Lindsey Frasier has already sent out a change notice about it to all stakeholders: https://groups.google.com/a/redhat.com/g/aos-devel/c/VO1qDq6OdBk/m/XM7ynrxAAAAJ.

I think that @eromanova97 has already requested the redirects [in the register](https://docs.google.com/spreadsheets/d/1KH2_p54Fr2KRFTUGzQdpqExox7iL9RM4_yiPNQpArjk/edit#gid=0).

**Comment by eromanova97** [comment]
Hello @mburke5678 @bburt-rh , yes, I did request the redirects already and the PR for that should be on the way soon. I know it is better to wait for the redirect PR to be ready, but because of upcoming merge conflicts, I would rather start working on the manual cherry-picks as soon as possible. The redirect PR would not get merged before those are fixed anyway. @mburke5678 I hope that answers your questions. Thank you!

**Pull request merged**